src/base/bittorrent/private/filterparserthread.cpp

```diff
@@ -43,33 +43,61 @@ namespace libt = libtorrent;
 
 namespace
 {
-    bool parseIPAddress(QString _ip, libt::address &address)
-    {
-        _ip = _ip.trimmed();
-
-        // Emule .DAT files contain leading zeroes in IPv4 addresses
-        // eg 001.009.106.186
-        // We need to remove them because both QHostAddress and Boost.Asio fail to parse them.
-        QStringList octets = _ip.split('.', QString::SkipEmptyParts);
-        if (octets.size() == 4) {
-            QString octet; // it is faster to not recreate this object in the loop
-            for (int i = 0; i < 4; i++) {
-                octet = octets[i];
-                if ((octet[0] == QChar('0')) && (octet.count() > 1)) {
-                    if ((octet[1] == QChar('0')) && (octet.count() > 2))
-                        octet.remove(0, 2);
-                    else
-                        octet.remove(0, 1);
-
-                    octets[i] = octet;
+    class DATIPv4Parser {
+    public:
+        const char* tryParse(const char* str, bool& ok) {
+            unsigned char number = 0;
+
+            const char* numberStart = str;
+            char* endptr;
+            for (; *str; ++str) {
+                if (*str == '.') {
+                    m_buf[number++] = static_cast<unsigned char>(strtol(numberStart, &endptr, 10));
+                    if (endptr != str) {
+                        break;
+                    }
+                    if (number == 4) { // an IP might end with '.': 192.168.1.2.
+                        ok = true;
+                        return str + 1;
+                    }
+                    numberStart = str + 1;
+                }
+            }
+#if 1
+            // The following is needed for parsing of a string with IP, but in the eMule files there is always a space after an IP,
+            // and this case is handled above
+            if (str != numberStart) {
+                m_buf[number] = static_cast<unsigned char>(strtol(numberStart, &endptr, 10));
+                if (endptr == str && number == 3) {
+                    ok = true;
+                    return str + 1;
                 }
             }
+#endif
 
-            _ip = octets.join(".");
+            ok = false;
+            return str + 1;
         }
 
+        libt::address_v4::bytes_type parsed() const {
+            return m_buf;
+        }
+
+    private:
+        libt::address_v4::bytes_type m_buf;
+    };
+
+    bool parseIPAddress(const QByteArray &_ip, libt::address &address)
+    {
+        DATIPv4Parser parser;
```

Comment:
On the second though this does not matter. Creating DATIPv4Parser is as simple as  incrementing the stack pointer. It is not surprising therefore that the code with local variable performs better.